### PR TITLE
Add SuppressedIssuesFileSynchronizer

### LINKED ISSUE #2889: Trigger suppressions file update whenever suppressions are fetched
### Description

Whenever suppressions are changed, call the service that is responsible for saving the suppressions to disk.

Relevant classes:
- `ISuppressedIssuesMonitor.SuppressionsUpdateRequested`
- `ISonarQubeIssuesProvider`

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Fixes #2889

### CODE REVIEW

File: src/Integration/Suppression/SuppressedIssuesFileSynchronizer.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    /// Responsible for listening to <see cref="ISuppressedIssuesMonitor.SuppressionsUpdateRequested"/> and calling
+    /// <see cref="ISuppressedIssuesFileStorage.Update"/> with the new suppressions.
+    /// </summary>
+    public interface ISuppressedIssuesFileSynchronizer : IDisposable
+    {
+    }
+
+    [Export(typeof(ISuppressedIssuesFileStorage))]
+    // todo: temp, will be replaced with real implementation
+    internal class DummySuppressedIssuesFileStorage : ISuppressedIssuesFileStorage
```
Comment (rita-sonarsource) [inline src/Integration/Suppression/SuppressedIssuesFileSynchronizer.cs:41]: Dummy implementation so the MEF import won't fail, this would be replaced by @ugras-ergun-sonarsource 's class

File: src/Integration/Suppression/SuppressedIssuesFileSynchronizer.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        }
+
+        public IEnumerable<SonarQubeIssue> Get(string sonarProjectKey)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    [Export(typeof(ISuppressedIssuesFileSynchronizer))]
+    internal class SuppressedIssuesFileSynchronizer : ISuppressedIssuesFileSynchronizer
```
Comment (rita-sonarsource) [inline src/Integration/Suppression/SuppressedIssuesFileSynchronizer.cs:54]: will make sealed to fix the SC issue after the PR comments

File: src/Integration/Suppression/SuppressedIssuesFileSynchronizer.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+{
+    /// <summary>
+    /// Responsible for listening to <see cref="ISuppressedIssuesMonitor.SuppressionsUpdateRequested"/> and calling
+    /// <see cref="ISuppressedIssuesFileStorage.Update"/> with the new suppressions.
+    /// </summary>
+    public interface ISuppressedIssuesFileSynchronizer : IDisposable
+    {
+    }
+
+    [Export(typeof(ISuppressedIssuesFileStorage))]
```
Comment (duncanp-sonar) [inline src/Integration/Suppression/SuppressedIssuesFileSynchronizer.cs:39]: It's not part of this change, but weren't we going to call this something more general like `IRoslynSettingsFileStorage`?

File: src/Integration/Suppression/SuppressedIssuesFileSynchronizer.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    [Export(typeof(ISuppressedIssuesFileSynchronizer))]
+    internal class SuppressedIssuesFileSynchronizer : ISuppressedIssuesFileSynchronizer
+    {
+        private readonly ISuppressedIssuesMonitor suppressedIssuesMonitor;
+        private readonly ISonarQubeIssuesProvider suppressedIssuesProvider;
+        private readonly ISuppressedIssuesFileStorage suppressedIssuesFileStorage;
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+
+        [ImportingConstructor]
+        public SuppressedIssuesFileSynchronizer(ISuppressedIssuesMonitor suppressedIssuesMonitor,
```
Comment (duncanp-sonar) [inline src/Integration/Suppression/SuppressedIssuesFileSynchronizer.cs:62]: Does this type need to be in `Integration`? It's specific to Roslyn suppressions, and all of the types it imports seem to be in `Core`. Could it be in the new suppressions assembly instead?

Comment (rita-sonarsource) [reply src/Integration/Suppression/SuppressedIssuesFileSynchronizer.cs:62]: The new assembly doesn't reference MEF so it can't be there. Additionally it's only used on the VS side, and that's where all the implementations of these interfaces are.

File: src/Integration.UnitTests/Suppression/SuppressedIssuesFileSynchronizerTests.cs [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        }
+
+        private SonarQubeIssue CreateSonarQubeIssue()
+        {
+            return new SonarQubeIssue(Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(),
+                Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), false,
+                SonarQubeIssueSeverity.Blocker, new DateTimeOffset(), new DateTimeOffset(), null, null);
+        }
+
+        private static BindingConfiguration CreateBindingConfiguration(string projectKey)
```
Comment (duncanp-sonar) [inline src/Integration.UnitTests/Suppression/SuppressedIssuesFileSynchronizerTests.cs:133]: Might be better to call this `CreateBoundConfiguration`/`CreateConnectedConfiguration` since it's hard-coded to return use connected mode.

**Review by duncanp-sonar — COMMENTED** [review]
Looks good, just a question about whether this can in the suppressions assembly instead.

**Review by duncanp-sonar — APPROVED** [review]
